_posts/2021-06-10-solidity-0.8.5-release-announcement.md

```diff
@@ -0,0 +1,221 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.5 Release Announcement'
+date: '2021-06-10'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.5](https://github.com/ethereum/solidity/releases/tag/v0.8.5) allows conversions
+from ``bytes`` to ``bytesNN`` values, adds the ``verbatim`` builtin function to inject
+arbitrary bytecode in Yul and fixes several smaller bugs.
+
+
+## Notable New Features
+### Bytes Conversion
+
+_Find the complete feature documentation [here](https://docs.soliditylang.org/en/v0.8.5/types.html#explicit-conversions)._
+
+This release introduces the ability to convert ``bytes`` and ``bytes`` slices to fixed bytes types ``bytes1`` / ... / ``bytes32``. While conversion between fixed-length bytes types has always been possible, it is now also possible to convert dynamically-sized bytes types to fixed-length bytes types.
+
+In case a byte array is longer than the target fixed bytes type, it will be truncated at the end:
+
+```solidity
+function f(bytes calldata c) public view returns (bytes8) {
+	// If c is longer than 8 bytes, truncation happens
+	return bytes8(c); 
+}
+```
+Calling ``f("12345678")`` will return ``"12345678"``, as will calling it as ``f("1234567890")``.
+If the array is shorter than the target fixed type, it will be padded with zeros at the end, so
+calling ``f("1234")`` will return ``"1234"``.
+
+A nice example of using the `bytes` conversion feature would be its application in proxies:
+
+```solidity
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.5;
+contract Proxy {
+	/// @dev Address of the client contract managed by proxy i.e., this contract
+	address client;
+	constructor(address _client) {
+		client = _client;
+	}
+	/// Forward call to "setOwner(address)" that is implemented by client
+	/// after doing basic validation on the address argument.
+	function forward(bytes calldata _payload) external {
+		require(_payload.length >= 4);
+		bytes4 sig = bytes4(_payload[:4]);
+		if (sig == bytes4(keccak256("setOwner(address)"))) {
+			address owner = abi.decode(_payload[4:], (address));
+			require(owner != address(0), "Address of owner cannot be zero.");
```

Comment:
Actually shouldn't we use user-defined errors instead?

Reply:
Probably... In that case, this part of documentation needs update as well, but I guess can be done later